components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/message/OAuthMessage.java

```diff
@@ -0,0 +1,253 @@
+import org.wso2.carbon.identity.oauth.endpoint.exception.InvalidRequestParentException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class OAuthMessage {
+
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+    private Map<String, Object> properties = new HashMap();
+    private OAuthAuthorizeState requestType;
+
+    private SessionDataCacheKey cacheKey;
+    private SessionDataCacheEntry resultFromLogin = null;
+    private SessionDataCacheEntry resultFromConsent = null;
+    private SessionDataCacheEntry sessionDataCacheEntry = null;
+
+    private boolean forceAuthenticate = false;
+    private boolean isPassiveAuthentication = false;
+    String sessionDataKeyFromConsent;
+
+    private OAuthMessage(HttpServletRequest request, HttpServletResponse response) {
+        this.request = request;
+        this.response = response;
+
+        String sessionDataKeyFromLogin = getSessionDataKey(request);
+        sessionDataKeyFromConsent = request.getParameter(OAuthConstants.SESSION_DATA_KEY_CONSENT);
+
+        if (StringUtils.isNotEmpty(sessionDataKeyFromLogin)) {
+            cacheKey = new SessionDataCacheKey(sessionDataKeyFromLogin);
+            resultFromLogin = SessionDataCache.getInstance().getValueFromCache(cacheKey);
+        }
+        if (StringUtils.isNotEmpty(sessionDataKeyFromConsent)) {
+            cacheKey = new SessionDataCacheKey(sessionDataKeyFromConsent);
+            resultFromConsent = SessionDataCache.getInstance().getValueFromCache(cacheKey);
+            SessionDataCache.getInstance().clearCacheEntry(cacheKey);
+        }
+    }
+
+    public Object getProperty(String key) {
+        if (properties != null) {
+            return properties.get(key);
+        } else {
+            return null;
+        }
+    }
+
+    public Map<String, Object> getProperties() {
+        return properties;
+    }
+
+    public void setProperty(String key, Object value) {
+        properties.put(key, value);
+    }
+
+    public void removeProperty(String key) {
+        properties.remove(key);
+    }
+
+    public HttpServletRequest getRequest() {
+        return request;
+    }
+
+    public HttpServletResponse getResponse() {
+        return response;
+    }
+
+    public OAuthAuthorizeState getRequestType() {
+        return requestType;
+    }
+
+    private void setRequestType(OAuthAuthorizeState requestType) {
+        this.requestType = requestType;
+        if (OAuthAuthorizeState.AUTHENTICATION_RESPONSE.equals(requestType)) {
+            setSessionDataCacheEntry(resultFromLogin);
+        } else if (OAuthAuthorizeState.USER_CONSENT_RESPONSE.equals(requestType)) {
+            setSessionDataCacheEntry(resultFromConsent);
+        }
+    }
+
+    public String getClientId() {
+        return request.getParameter("client_id");
```

Comment:
Use a defined a constant for "client_id"

Reply:
done